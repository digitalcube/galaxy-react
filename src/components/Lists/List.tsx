import React, { FC, PropsWithChildren } from 'react';
import { useMemo } from 'react';

export type ListProps = PropsWithChildren<{
  className?: string;
  rounded?: boolean;
}>;

export const List: FC<ListProps> = ({
  className,
  rounded = false,
  children,
}) => {
  const classNames = useMemo(() => {
    return [
      'base-notification-box',
      'bg-white',
      rounded ? 'rounded' : '',
      className,
    ].filter(Boolean);
  }, [rounded, className]);
  return <div className={classNames.join(' ')}>{children}</div>;
};
