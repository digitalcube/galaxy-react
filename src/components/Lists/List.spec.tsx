import { render } from '@testing-library/react';
import React from 'react';
import { Default as List } from './List.stories';

describe('List', () => {
  it('to match snapshot(default)', () => {
    const { asFragment } = render(<List {...(List.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
