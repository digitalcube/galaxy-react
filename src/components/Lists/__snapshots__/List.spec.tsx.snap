// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List to match snapshot(default) 1`] = `
<DocumentFragment>
  <div
    class="shifter-dashboard-columns pt-3"
  >
    <div
      class="base-notification-box rounded"
    >
      <div
        class="notification d-flex px-4 py-3 rounded align-items-center justify-content-between"
      >
        <div
          class="notification-status"
        >
          test 
        </div>
        <div
          class="custom-control custom-switch c-toggle-switch"
        >
          <input
            checked=""
            class="custom-control-input"
            id="dummy-toggle"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="dummy-toggle"
          >
            <span
              class="d-none"
            >
              dummy
            </span>
          </label>
        </div>
      </div>
      <div
        class="notification d-flex px-4 py-3 rounded align-items-center justify-content-between"
      >
        <div
          class="notification-status"
        >
          test 
        </div>
        <button
          class="btn-colored btn-smaller btn btn-secondary"
        >
          Button
        </button>
      </div>
      <div
        class="notification d-flex px-4 py-3 rounded align-items-center justify-content-between"
      >
        <div
          class="notification-status"
        >
          test 
          <span
            class="c-toggle-switch-status active font-weight-bold"
          >
            on
          </span>
        </div>
      </div>
      <div
        class="notification d-flex px-4 py-3 rounded align-items-center justify-content-between"
      >
        <div
          class="notification-status"
        >
          test 
          <span
            class="c-toggle-switch-status inactive font-weight-bold"
          >
            disabled
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
